com.avaloq.tools.ddk.xtext.expression/src-gen/com/avaloq/tools/ddk/xtext/expression/expression/BooleanLiteral.java

```diff
@@ -21,26 +21,26 @@
  */
 public interface BooleanLiteral extends Literal
 {
-  /**
-   * Returns the value of the '<em><b>Val</b></em>' attribute.
-   * <!-- begin-user-doc -->
-   * <!-- end-user-doc -->
-   * @return the value of the '<em>Val</em>' attribute.
-   * @see #setVal(String)
-   * @see com.avaloq.tools.ddk.xtext.expression.expression.ExpressionPackage#getBooleanLiteral_Val()
-   * @model
-   * @generated
-   */
-  String getVal();
+	/**

```

Comment:
This (and several others in this plugin) appears to just contain whitespace changes that replace 2-spaces with 1-tab. Has the configuration for something changed?

Reply:
My findings are the following:
- In ASMD and other plugins we have tabs, so I think this plug-in was not generated in a very long time, and the code generator behaviour has changed, why I do not know.
- The generator should pick the project formatter settings https://www.eclipse.org/forums/index.php/t/1065524/, I checked our project settings and we have org.eclipse.jdt.core.formatter.tabulation.char=space
org.eclipse.jdt.core.formatter.tabulation.size=2, org.eclipse.jdt.core.formatter.use_tabs_only_for_leading_indentations=true. I have set the later to false and restarted eclipse, just in case, and it did not do any difference
- The generator uses internally a JMerge https://www.eclipse.org/modeling/emf/docs/2.x/whatsnew/merge2.3.html, which offers the possibility to be configured, but I could not figure out how. I tried to specify in the genmodel the property codeFormatting="true" but it did not do any difference.
- The default for JMerge should be actually spaces, the file /templates/emf-merge.xml in the org.eclipse.emf.codegen.ecore plug-in has the following

<merge:options 
  indent="  "  
  braceStyle="matching"  
  redirect="Gen"
  block="\s*@\s*generated\s*NOT\s*(?:\n\r?|\r\n?)"
  noImport="\s*//\s*import\s+([\w.*]*)\s*;\s*(?:\n\r?|\r\n?)"
  xmlns:merge="http://www.eclipse.org/org/eclipse/emf/codegen/jmerge/Options">

<!-- Tabbed Standard

  indent="&#x9;"  
  braceStyle="standard"  

-->
- Finally, I got it to generate spaces by overriding the creation of the JControlModel, I will show you this later, but it does not look as if this is supposed to be customized like that.